src/Neo/SmartContract/StorageItem.cs

```diff
@@ -181,14 +182,9 @@ public void Set(BigInteger integer)
             _value = null;
         }
 
-        /// <summary>
-        /// Sets the interoperable value of the storage.
-        /// </summary>
-        /// <param name="interoperable">The <see cref="IInteroperable"/> value of the <see cref="StorageItem"/>.</param>
-        public void Set(IInteroperable interoperable)
+        internal static void AssertIsSerializable(IInteroperable interoperable)
         {
-            _cache = interoperable;
-            _value = null;
+            BinarySerializer.Serialize(interoperable.ToStackItem(null), ExecutionEngineLimits.Default);
```

Comment:
@shargon 

This is the most significant changes in this PR, right?

Any existing code that relied on implicitly converting a StorageItem instance to a BigInteger will now fail to compile.

Reply:
Only Interop items has the problem